flink-connector-opensearch/src/main/java/org/apache/flink/connector/opensearch/sink/OpensearchWriter.java

```diff
@@ -343,7 +346,11 @@ private void extractFailures(BulkRequest request, BulkResponse response) {
         if (chainedFailures == null) {
             return;
         }
-        throw new FlinkRuntimeException(chainedFailures);
+        if (failureHandler == null) {
```

Comment:
Dealing with`null` is better to be avoided: we could use the default `failureHandler`:

```java
private static final FailureHandler DEFAULT_FAILURE_HANDLER = ex -> throw new FlinkRuntimeException(ex);
```

and never deal with nullability / `if`s